graphrag/index/run/run.py

```diff
@@ -0,0 +1,265 @@
+    callbacks = callbacks or _create_reporter(config.reporting, root_dir)
+    dataset = (
+        dataset
+        if dataset is not None
+        else await _create_input(config.input, progress_reporter, root_dir)
+    )
+
+    if is_update_run:
+        # TODO: Filter dataset to only include new data (this should be done in the input module)
+        pass
+    post_process_steps = input_post_process_steps or _create_postprocess_steps(
+        config.input
+    )
+    workflows = workflows or config.workflows
+
+    if dataset is None:
+        msg = "No dataset provided!"
+        raise ValueError(msg)
+
+    async for table in run_pipeline(
+        workflows=workflows,
+        dataset=dataset,
+        storage=storage,
+        cache=cache,
+        callbacks=callbacks,
+        input_post_process_steps=post_process_steps,
+        memory_profile=memory_profile,
+        additional_verbs=additional_verbs,
+        additional_workflows=additional_workflows,
+        progress_reporter=progress_reporter,
+        emit=emit,
+        is_resume_run=is_resume_run,
+    ):
+        yield table
+
+
+async def run_pipeline(
+    workflows: list[PipelineWorkflowReference],
+    dataset: pd.DataFrame,
+    storage: PipelineStorage | None = None,
+    cache: PipelineCache | None = None,
+    callbacks: WorkflowCallbacks | None = None,
+    progress_reporter: ProgressReporter | None = None,
+    input_post_process_steps: list[PipelineWorkflowStep] | None = None,
+    additional_verbs: VerbDefinitions | None = None,
+    additional_workflows: WorkflowDefinitions | None = None,
+    emit: list[TableEmitterType] | None = None,
+    memory_profile: bool = False,
+    is_resume_run: bool = False,
+    **_kwargs: dict,
+) -> AsyncIterable[PipelineRunResult]:
+    """Run the pipeline.
+
+    Args:
+        - workflows - The workflows to run
+        - dataset - The dataset to run the pipeline on, specifically a dataframe with the following columns at a minimum:
+            - id - The id of the document
+            - text - The text of the document
+            - title - The title of the document
+            These must exist after any post process steps are run if there are any!
+        - storage - The storage to use for the pipeline
+        - cache - The cache to use for the pipeline
+        - reporter - The reporter to use for the pipeline
+        - input_post_process_steps - The post process steps to run on the input data
+        - additional_verbs - The custom verbs to use for the pipeline
+        - additional_workflows - The custom workflows to use for the pipeline
+        - debug - Whether or not to run in debug mode
+    Returns:
+        - output - An iterable of workflow results as they complete running, as well as any errors that occur
+    """
+    start_time = time.time()
+    stats = PipelineRunStats()
+    storage = storage or MemoryPipelineStorage()
+    cache = cache or InMemoryCache()
```

Comment:
Could we access these 3 variables from the context declared below and move the initialization of these to the _create_run_context line? that way we remove variables from here and make sure we are using the right ones from the context

Reply:
Done